lib/spack/spack/package.py

```diff
@@ -1217,6 +1236,15 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class PythonExtension(Package):
+    def setup_dependent_environment(self, env, dependent_spec):
+        pass
+
+    def setup_environment(self, env):
+        site_packages = glob.glob(join_path(self.spec.prefix.lib, "python*/site-packages"))
+        if site_packages:
+            env.prepend_path('PYTHONPATH', site_packages[0])
```

Comment:
@tgamblin Ok, sorry for the noise, but I think I may have it. What if we patch the `setup_environment` and `setup_dependent_environment` of python extensions from:
```
def modify_module(self, module, spec, dependent_spec):
...
```
of the `python` package, if:
- the functions were not overridden explicitly by `dependent_spec`
- `dependent_spec` **extends** python 

After all `modify_module` seems the right place to do those kind of things...